topic-operator/src/main/java/io/strimzi/operator/topic/v2/TopicOperatorConfig.java

```diff
@@ -0,0 +1,247 @@
+        }
+
+        if (this.securityProtocol().equals("SASL_SSL") || this.securityProtocol().equals("SSL") || this.tlsEnabled()) {
+            kafkaClientProps.put(SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG, this.sslEndpointIdentificationAlgorithm());
+
+            if (!this.truststoreLocation().isEmpty()) {
+                kafkaClientProps.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, this.truststoreLocation());
+            }
+
+            if (!this.truststorePassword().isEmpty()) {
+                if (this.truststoreLocation().isEmpty()) {
+                    throw new InvalidConfigurationException("TLS_TRUSTSTORE_PASSWORD was supplied but TLS_TRUSTSTORE_LOCATION was not supplied");
+                }
+                kafkaClientProps.put(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, this.truststorePassword());
+            }
+
+            if (!this.keystoreLocation().isEmpty() && !this.keystorePassword().isEmpty()) {
+                kafkaClientProps.put(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, this.keystoreLocation());
+                kafkaClientProps.put(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, this.keystorePassword());
+            }
+        }
+
+        if (this.saslEnabled()) {
+            putSaslConfigs(kafkaClientProps);
+        }
+        return kafkaClientProps;
+    }
+
+    private void putSaslConfigs(Map<String, Object> kafkaClientProps) {
+        TopicOperatorConfig config = this;
+        String saslMechanism;
+        String jaasConfig;
+        String username = config.saslUsername();
+        String password = config.saslPassword();
+        String configSaslMechanism = config.saslMechanism();
+
+        if (username.isEmpty() || password.isEmpty()) {
+            throw new InvalidConfigurationException("SASL credentials are not set");
+        }
+
+        if ("plain".equals(configSaslMechanism)) {
+            saslMechanism = "PLAIN";
+            jaasConfig = "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"" + username + "\" password=\"" + password + "\";";
+        } else if ("scram-sha-256".equals(configSaslMechanism) || "scram-sha-512".equals(configSaslMechanism)) {
+            jaasConfig = "org.apache.kafka.common.security.scram.ScramLoginModule required username=\"" + username + "\" password=\"" + password + "\";";
+
+            if ("scram-sha-256".equals(configSaslMechanism)) {
+                saslMechanism = "SCRAM-SHA-256";
+            } else {
+                saslMechanism = "SCRAM-SHA-512";
+            }
+        } else {
+            throw new IllegalArgumentException("Invalid SASL_MECHANISM type: " + configSaslMechanism);
+        }
+
+        kafkaClientProps.put(SaslConfigs.SASL_MECHANISM, saslMechanism);
+        kafkaClientProps.put(SaslConfigs.SASL_JAAS_CONFIG, jaasConfig);
+    }
+
+    @Override
+    public String toString() {
+        String mask = "********";
+        return "TopicOperatorConfig{" +
```

Comment:
@scholzj did we decide multi-line log messages were fine? Should I change this to reflect that?

Reply:
Nobody protested -> so I would say that yes, they are fine.